cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -672,6 +698,146 @@ public Route generateExternalBootstrapRoute() {
+        }
+
+        return null;
+    }
+
+    /**
+     * Generates a bootstrap ingress which can be used to bootstrap clients outside of Kubernetes.
+     * @return The generated Ingress
+     */
+    public Ingress generateExternalBootstrapIngress() {
+        if (isExposedWithIngress()) {
+            KafkaListenerExternalIngress listener = (KafkaListenerExternalIngress) listeners.getExternal();
+            Map<String, String> dnsAnnotations;
+            String host;
+
+            if (listener.getConfiguration() != null && listener.getConfiguration().getBootstrap() != null && listener.getConfiguration().getBootstrap().getHost() != null) {
+                host = listener.getConfiguration().getBootstrap().getHost();
+                dnsAnnotations = listener.getConfiguration().getBootstrap().getDnsAnnotations();
+            } else {
+                throw new InvalidResourceException("Boostrap hostname is required for exposing Kafka cluster using Ingress");
+            }
+
+            HTTPIngressPath path = new HTTPIngressPathBuilder()
+                    .withPath("/")
+                    .withNewBackend()
+                        .withNewServicePort(EXTERNAL_PORT)
+                        .withServiceName(externalBootstrapServiceName(cluster))
+                    .endBackend()
+                    .build();
+
+            IngressRule rule = new IngressRuleBuilder()
+                    .withHost(host)
+                    .withNewHttp()
+                        .withPaths(path)
+                    .endHttp()
+                    .build();
+
+            IngressTLS tls = new IngressTLSBuilder()
+                    .withHosts(host)
+                    .build();
+
+            Ingress ingress = new IngressBuilder()
+                    .withNewMetadata()
+                        .withName(serviceName)
+                        .withLabels(getLabelsWithName(serviceName, templateExternalBootstrapIngressLabels))
+                        .withAnnotations(mergeAnnotations(generateInternalIngressAnnotations(), templateExternalBootstrapIngressAnnotations, dnsAnnotations))
+                        .withNamespace(namespace)
+                        .withOwnerReferences(createOwnerReference())
+                    .endMetadata()
+                    .withNewSpec()
+                        .withRules(rule)
+                        .withTls(tls)
+                    .endSpec()
+                    .build();
+
+            return ingress;
+        }
+
+        return null;
+    }
+
+    /**
+     * Generates the annotations needed to configure the Ingress as TLS passthrough
+     *
+     * @return  Map with the annotations
+     */
+    private Map<String, String> generateInternalIngressAnnotations() {
+        Map<String, String> internalAnnotations = new HashMap<>(4);
+        internalAnnotations.put("kubernetes.io/ingress.class", "nginx");
+        internalAnnotations.put("ingress.kubernetes.io/ssl-passthrough", "true");
```

Comment:
Are these supported on all the kube versions we support?

Reply:
This is independent on Kubernetes version. This is more bound to the Ingress. The way this works, is that Kubernetes provides the Ingress API. But you have to install the Ingress controller which will manage this (the first operator? ;-)). The used most commonly is the Nginx controller which I tested this with and which is targeted by these annotations.

Reply:
I see my questions are doing a good job of exposing my ignorance about this stuff. Thanks for taking the time to explain.

Reply:
No worries.